### PR TITLE
Add pause and resume feature for focus timer

### PR DESCRIPTION
**Pull request by JingHaoooo**
satisfies #69, #78 and #80

### CODE REVIEW

File: src/main/java/focus/CountdownTimer.java [hunk 1]
```diff
@@ -19,8 +19,7 @@ public class CountdownTimer {
     private int minutes;
     private int seconds;
     private int inputMinutes;
-    private boolean isStarted;
-    private AtomicBoolean isCompleted;
+    private AtomicBoolean isStarted;
```
Comment (damiwee) [inline src/main/java/focus/CountdownTimer.java:22]: I like how you followed naming conventions for boolean variables

File: src/main/java/focus/CountdownTimer.java [hunk 2]
```diff
@@ -71,22 +69,40 @@ public void run() {
...
+    public void setResume() {
+        isRunning.set(true);
+        Ui.printMessageWithSepNewLine("Countdown timer resumed.");
+    }
+
     public boolean getRunningStatus() {
+        return isStarted.get();
+    }
+
+    public boolean getPausedStatus() {
```
Comment (damiwee) [inline src/main/java/focus/CountdownTimer.java:105]: I like how you used SLAP to improve code quality

File: src/main/java/focus/CountupTimer.java [hunk 3]
```diff
@@ -11,27 +11,56 @@ public class CountupTimer {
...
     }
 
+    public void setPause() {
+        LocalDateTime currentTime = LocalDateTime.now();
+        Duration timeElapsed = Duration.between(startTiming, currentTime);
+        totalHours += timeElapsed.toHours();
+        totalMinutes += timeElapsed.toMinutes() % MINUTES_DIVISION;
+        totalSeconds += timeElapsed.toSeconds() % MINUTES_DIVISION;
+        isPaused = true;
+        Ui.printMessageWithSepNewLine("Count up timer paused \n" +
```
Comment (damiwee) [inline src/main/java/focus/CountupTimer.java:50]: Perhaps consider adding line break here to improve readability (nitpicking)

File: src/main/java/focus/FocusTimer.java [hunk 4]
```diff
@@ -33,18 +33,43 @@ public void setStopTiming() {
...
+        }
+    }
+
+    public void setResumeTiming() {
+        if (timerMode) {
+            countdownTimer.setResume();
+        } else {
+            countupTimer.setResume();
+        }
+    }
```
Comment (damiwee) [inline src/main/java/focus/FocusTimer.java:74]: I like how you used SLAP here to improve code quality too.

File: src/main/java/commands/focuscommands/SetResumeCommand.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    public FocusTimer focusTimer;
+
+    public SetResumeCommand(FocusTimer timer) {
+        this.focusTimer = timer;
+    }
+
+    public void execute() throws FocusException {
+        if (!focusTimer.getPausedStatus() || !focusTimer.getStatus()) {
+            throw new FocusException("Timer is already resumed or Timer hasn't started..");
+        }
```
[the remark this answers is not in the document]
Comment (JingHaoooo) [reply src/main/java/commands/focuscommands/SetResumeCommand.java:17]: Thanks for the suggestion. Will refactor the code to improve readability.

**Comment by damiwee** [comment]
Looks good to me. Great effort in making the changes!

**Pull request merged**